pkg/client/clientcmd/client_config.go

```diff
@@ -250,6 +250,9 @@ func (config DirectClientConfig) Namespace() (string, error) {
 	if len(configContext.Namespace) == 0 {
 		return api.NamespaceDefault, nil
 	}
+	if configContext.Namespace == "*" {
```

Comment:
----- Original Message -----
> > @@ -250,6 +250,9 @@ func (config DirectClientConfig) Namespace() (string,
> > error) {
> >  	if len(configContext.Namespace) == 0 {
> >  		return api.NamespaceDefault, nil
> >  	}
> > +	if configContext.Namespace == "*" {
> 
> > I don't know that we want to allow namespace to be stored in the config as
> > *. That violates some of the fundamentals of our context settings.
> 
> It means only config file such as `$HOME/.kube/config` or both config file
> and argument? I assume the latter.
> 
> 
> 
> > Perhaps we can add a boolean 'selectAll' which is used if the namespace is
> > empty?
> 
> IIUC, empty namespace is handled as _default_ namespace, can it change?

No, we never want namespace to default to all.  I don't think there is value in config defaulting to all namespaces, so I don't really want to change the config to support something different.  Namespaces are a context - cross namespace is an exceptional operation.  Our tools are designed to work within a namespace, and be able to work across namespaces in certain cases.

> Or,  how about adding something like `--all-namespace` for argument and
> `allNamespace: true|false` option for config file to handle all namespace.

I'd rather not add a new flag if --namespace='*' works for now.  I don't think this is commonly used enough be worth the cognitive overhead of another flag.

> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7024/files#r28874089